### PR TITLE
Refactoring archive testing to a single yml file.

### PR DESCRIPTION
**Pull request by ryanwilson**
This reduces the duplication in each of the workflow files. This also
will make archive testing with SwiftPM more straightforward.

### CODE REVIEW

File: .github/workflows/archiving.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  schedule:
+    # Run every day at 2am (PST) - cron uses UTC times
+    # This is set to 3 hours after zip workflow finishes so zip testing can run after.
+    - cron:  '0 10 * * *'
+
+jobs:
+  # Archive tests for pods that support iOS only.
+  pods-ios-only-cron:
+    # Don't run on private repo.
+    if: github.event_name == 'schedule' && github.repository != 'FirebasePrivate/firebase-ios-sdk'
```
Comment (paulb777) [inline .github/workflows/archiving.yml:13]: Would you comment this to test?

File: .github/workflows/archiving.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    - name: Setup Bundler
+      run: scripts/setup_bundler.sh
+    - name: Setup project and archive
+      run: scripts/test_archiving.sh ${{ matrix.pod }} ios ArchiveOutputs/ios.xcarchive
+
+
+  # Archive tests for pods that support iOS, tvOS, and macOS.
+  pods-ios-tvos-macos-cron:
+    # Don't run on private repo.
+    if: github.event_name == 'schedule' && github.repository != 'FirebasePrivate/firebase-ios-sdk'
```
Comment (paulb777) [inline .github/workflows/archiving.yml:34]: And here

**Review by paulb777 — COMMENTED** [review]
Much more maintainable. Thanks!

**Comment by paulb777** [comment]
remoteconfig test failure is addressed by #6514

**Comment by ryanwilson** [comment]
> Probably shouldn't be added here

Thanks, got caught with a `git commit -a` 😝  will remove.

**Review by paulb777 — APPROVED** [review]

**Pull request merged**